File: task-manager-api/Entities/Task.cs
using System;
using System.Collections.Generic;

namespace TaskManagerServer.Entities
{
    public class Task
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public int? TaskRegisterID { get; set; }
        public DateTime? Deadline { get; set; }
        public DateTime DateOfCreation { get; set; }
        public bool IsDone { get; set; }
        public bool Reminder { get; set; }
        public bool IsInArchive { get; set; }
        public int UserID { get; set; }
        public IEnumerable<Tag> Tags { get; set; }
    }
}

File: task-manager-client/src/app/models/task.ts
export class Task {
    ID: number;
    Name: string;
    TaskRegisterID: number;
    Deadline: Date;
    DateOfCreation: Date;
    IsDone: boolean;
    Reminder: boolean;
    IsInArchive: boolean;
    UserID: number;
    contructor(
        ID: number = null,
        Name: string = '',
        TaskRegisterID: number = null,
        Deadline: Date = null,
        DateOfCreation: Date = null,
        IsDone: boolean = null,
        Reminder: boolean = null,
        IsInArchive: boolean = null,
        UserID: number = null,
    )
    {
        this.ID = ID;
        this.Name = Name;
        this.TaskRegisterID = TaskRegisterID;
        this.Deadline = Deadline;
        this.DateOfCreation = DateOfCreation;
        this.IsDone = IsDone;
        this.Reminder = Reminder;
        this.IsInArchive = IsInArchive;
        this.UserID = UserID;
    }
}
